### ISSUE TITLE
[dedup] Use APFS clone (CoW) on macOS

### ISSUE THREAD

**Issue by boozook** [pkolaczk/fclones#219]
Bug, kind: critical, `dedup`-cmd broken on macOS.
Docs: `man clonefile`
Min macOS: 10.12 (Sierra)
FS required: APFS

__TL;DR: Use `cp -c` instead of `cp --reflink` on macOS.__

### Explaination:
`cp -c` uses `clonefile` syscall, flag `-c` overrides default `copy/duplicate` behaviour and clone files via clonefile() instead — see man `cp(1)`. __The behaviour is identical to that of the Linux `cp` flag `--reflink`__ but in macOS `cp` __have no `--reflink` parameter,__ so with `--reflink` you'll get an errors like `cp: illegal option...`.

- - -

## Update

Seems to there's not a problem with `dedupe`, but with `help` documentation and cli (interface).
- `fclones dedupe -h` tells this:
> `--dry-run` Don't perform any changes on the file-system, but writes a log of file operations to the standard output.

- `fclones dedupe --dry-run <some.txt` prints commands like this:
```
mv /path/A /path/B
cp --reflink=always /path/C /path/A
rm /path/B
```

So I was sure that fclones without `--dry-run` will execute that commands.

**Comment by kapitainsky** [comment]
wow - indeed `dedupe` on macOS on does nothing...  but interestingly does not throw any error neither.

It should be very easy PR to fix it

**Comment by boozook** [comment]
> indeed dedupe on macOS does nothing...
I've run with `dedupe --dry-run` and inspect output.

**Comment by kapitainsky** [comment]
but.... 4 x 10GB identical files

```
$ ls -lh
total 83886080
-rw-r--r--@ 1 kptsky  staff    10G Aug  7 22:21 10GB.bin
-rw-r--r--@ 1 kptsky  staff    10G Aug  7 22:21 10GB.bin.1
-rw-r--r--@ 1 kptsky  staff    10G Aug  7 22:22 10GB.bin.2
-rw-r--r--@ 1 kptsky  staff    10G Aug  7 22:22 10GB.bin.3

$ df -h /System/Volumes/Data
Filesystem     Size   Used  Avail Capacity iused      ifree %iused  Mounted on
/dev/disk1s1  932Gi  541Gi  332Gi    62% 2404443 3484848560    0%   /System/Volumes/Data

$ fclones group . | fclones dedupe
...
[2023-08-07 22:24:01.028] fclones:  info: Processed 3 files and reclaimed up to 32.2 GB space

$ ls -lh
total 83886080
-rw-r--r--@ 1 kptsky  staff    10G Aug  7 22:21 10GB.bin
-rw-r--r--@ 1 kptsky  staff    10G Aug  7 22:21 10GB.bin.1
-rw-r--r--@ 1 kptsky  staff    10G Aug  7 22:22 10GB.bin.2
-rw-r--r--@ 1 kptsky  staff    10G Aug  7 22:22 10GB.bin.3

$ df -h /System/Volumes/Data
Filesystem     Size   Used  Avail Capacity iused      ifree %iused  Mounted on
/dev/disk1s1  932Gi  511Gi  362Gi    59% 2404444 3799378320    0%   /System/Volumes/Data
```

there is no problem with (CoW) dedupe on APFS

there is problem what `--dry-run` shows

**Comment by boozook** [comment]
Great!
Well, so this can be closed or still open for docs/interface improvement.
Thank you so much!

**Comment by kapitainsky** [comment]
Leave it open:) something is not 100% right - at least with `--dry-run`

**Comment by kapitainsky** [comment]
I think I have identified the problem. PR already posted.
